### ISSUE TITLE
Support for DM21?

### ISSUE THREAD

**Issue by jchodera** [psi4/psi4#2380]
I realize it was only released recently, but is there any possibility of supporting the [DM21 functionals](https://github.com/deepmind/deepmind-research/tree/master/density_functional_approximation_dm21) ([paper](https://doi.org/10.1126/science.abj6511))?

**Comment by JonathonMisiewicz** [comment]
@susilehtola

**Comment by susilehtola** [comment]
The authors' DM21 functional has a PySCF interface, so you can already use PySCF to run calculations.

Hooking it up to Psi4 will be a much bigger task, since unlike PySCF, Psi4 doesn't handle density functionals in Python...

**Comment by jchodera** [comment]
> The authors' DM21 functional implementation has a PySCF interface, so you can already use PySCF to run calculations.

Indeed! Our workflows use [QCEngine](https://github.com/MolSSI/QCEngine), which does not yet support PySCF. It should be relatively straightforward to add, however---I've opened an issue at https://github.com/MolSSI/QCEngine/issues/335 to discuss this possbility.

> Hooking it up to Psi4 will be a much bigger task, since unlike PySCF, Psi4 doesn't handle density functionals in Python...

DM21 can be used from C++: the README describes this [here](https://github.com/deepmind/deepmind-research/tree/master/density_functional_approximation_dm21#using-dm21-from-c).
